File: trip-kit/src/main/java/com/skedgo/android/tripkit/RouteService.java
package com.skedgo.android.tripkit;

import android.support.annotation.NonNull;

import com.skedgo.android.common.model.Query;
import com.skedgo.android.common.model.TripGroup;

import java.util.List;

import rx.Observable;

public interface RouteService {
  @NonNull Observable<List<TripGroup>> routeAsync(@NonNull Query query);
}

File: TripKitAndroid/src/main/java/com/skedgo/android/tripkit/RouteService.kt
package com.skedgo.android.tripkit

import com.skedgo.android.common.model.Query
import com.skedgo.android.common.model.TripGroup
import rx.Observable

interface RouteService {
  @Deprecated(
      "Use GetA2bRoutingResults instead",
      ReplaceWith("GetA2bRoutingResults", "skedgo.tripkit.a2brouting.GetA2bRoutingResults"),
      DeprecationLevel.WARNING
  )
  fun routeAsync(query: Query): Observable<List<TripGroup>>
}
